### ISSUE TITLE
Replace the image getHeight and getWidth with a faster implementation

### ISSUE THREAD

**Issue by wezell** [dotCMS/core#7813]
We should replace the getHeight / getWidth impls in our FileAsset.java class with speedier, more memory friendly implementations that use the ImageReader class.  

https://github.com/dotCMS/core/blob/master-3.2/src/com/dotmarketing/portlets/fileassets/business/FileAsset.java#L90

see: http://stackoverflow.com/questions/672916/how-to-get-image-height-and-width-using-java for more details and some ad-hoc benchmarking.

This call is probably coming from a ContentMap being used on the front end that gets converted to a FileAsset we should probably cache the resulting converted FileAsset and its properties so we don't have to re-read the image to get the width/height every page view.

Reason for this change - we have a customer who is seeing a blocking lock in their thread dumps:
```
"http-80-247" daemon prio=10 tid=0x00007f69a81ce000 nid=0x5efb waiting for monitor entry [0x00007f697941a000]
   java.lang.Thread.State: BLOCKED (on object monitor)
	at sun.java2d.cmm.lcms.LCMS.getTagSize(Native Method)
	- waiting to lock <0x000000044b340a48> (a sun.java2d.cmm.lcms.LCMS)
	at java.awt.color.ICC_Profile.getData(ICC_Profile.java:1370)
	at java.awt.color.ICC_Profile.getData(ICC_Profile.java:1358)
	at java.awt.color.ICC_Profile.getNumComponents(ICC_Profile.java:1460)
	at java.awt.image.ColorConvertOp.filter(ColorConvertOp.java:481)
	at com.sun.imageio.plugins.jpeg.JPEGImageReader.acceptPixels(JPEGImageReader.java:1263)
	at com.sun.imageio.plugins.jpeg.JPEGImageReader.readImage(Native Method)
	at com.sun.imageio.plugins.jpeg.JPEGImageReader.readInternal(JPEGImageReader.java:1231)
	at com.sun.imageio.plugins.jpeg.JPEGImageReader.read(JPEGImageReader.java:1034)
	at javax.imageio.ImageIO.read(ImageIO.java:1448)
	at javax.imageio.ImageIO.read(ImageIO.java:1308)
	at com.dotmarketing.portlets.fileassets.business.FileAsset.getHeight(FileAsset.java:90)
	at sun.reflect.GeneratedMethodAccessor502.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
```

**Comment by jtesser** [comment]
@wezell do you have a pull request here?

**Comment by wezell** [comment]
I think it is this one?  https://github.com/dotCMS/core/pull/7815

**Comment by jtesser** [comment]
thats it.  I lost it lol

**Comment by jgambarios** [comment]
We still can do some teaks in order to improve this issue, please read the notes on the pull request: https://github.com/dotCMS/core/pull/7815

**Comment by rogelio-blanco** [comment]
@wezell do you know how can I test this?

**Comment by rogelio-blanco** [comment]
We need to add a recommendation in our [documentation page](http://dotcms.com/docs/latest/content-object-in-velocity), if we want to improve how the velocity objects are used, we need to encourage to cache velocity objects.   

Please see the example below:

Example 1, we are creating new instances of the "fileAsset" object, so each time you call fileAsset.XXX, behind the scenes is creating a new object each time.  **Not recommended**
```
#set($file = $dotcontent.find("4d7daefa-6adb-4b76-896d-c2d9f95b2280"))

uri: $file.fileAsset.uri
inode: $file.fileAsset.inode
identifier: $file.fileAsset.identifier
path: $file.fileAsset.path
mimeType: $file.fileAsset.mimeType
extension: $file.fileAsset.extension
fileSize: $file.fileAsset.fileSize
width: $file.fileAsset.width
height: $file.fileAsset.height
```

Example 2, we are creating just one instances of the "fileAsset" object and operate to the same object, this can improve the performance of the vtl.  **Recommended**
```
#set($file = $dotcontent.find("4d7daefa-6adb-4b76-896d-c2d9f95b2280"))

#set($fileAssetVar = $file.fileAsset)

uri: $fileAssetVar.uri
inode: $fileAssetVar.inode
identifier: $fileAssetVar.identifier
path: $fileAssetVar.path
mimeType: $fileAssetVar.mimeType
extension: $fileAssetVar.extension
fileSize: $fileAssetVar.fileSize
width: $fileAssetVar.width
height: $fileAssetVar.height
```

**Comment by rogelio-blanco** [comment]
Please ignore the comment of changing our documentation we are going to fix this in #8048

**Comment by bryanboza** [comment]
Fixed, tested on the last 3.2.4 PR // H2 // FF
